cluster/examples/kubernetes/ceph/monitoring/prometheus-ceph-v14-rules.yaml

```diff
@@ -8,329 +8,296 @@ metadata:
-            severity_level: warning
-            storage_type: ceph
-          expr: |
-            count(count(ceph_mon_metadata{job="rook-ceph-mgr"}) by (ceph_version)) > 1
-          for: 10m
-          labels:
-            severity: warning
-    - name: cluster-utilization-alert.rules
-      rules:
-        - alert: CephClusterNearFull
-          annotations:
-            description:
-              Storage cluster utilization has crossed 75% and will become read-only
-              at 85%. Free up some space or expand the storage cluster.
-            message:
-              Storage cluster is nearing full. Data deletion or cluster expansion
-              is required.
-            severity_level: warning
-            storage_type: ceph
-          expr: |
-            ceph_cluster_total_used_raw_bytes / ceph_cluster_total_bytes > 0.75
-          for: 5s
-          labels:
-            severity: warning
-        - alert: CephClusterCriticallyFull
-          annotations:
-            description:
-              Storage cluster utilization has crossed 80% and will become read-only
-              at 85%. Free up some space or expand the storage cluster immediately.
-            message:
-              Storage cluster is critically full and needs immediate data deletion
-              or cluster expansion.
-            severity_level: error
-            storage_type: ceph
-          expr: |
-            ceph_cluster_total_used_raw_bytes / ceph_cluster_total_bytes > 0.80
-          for: 5s
-          labels:
-            severity: critical
-        - alert: CephClusterReadOnly
-          annotations:
-            description:
-              Storage cluster utilization has crossed 85% and will become read-only
-              now. Free up some space or expand the storage cluster immediately.
-            message:
-              Storage cluster is read-only now and needs immediate data deletion
-              or cluster expansion.
-            severity_level: error
-            storage_type: ceph
-          expr: |
-            ceph_cluster_total_used_raw_bytes / ceph_cluster_total_bytes >= 0.85
-          for: 0s
-          labels:
-            severity: critical
+  - name: ceph.rules
+    rules:
+    - expr: |
+        kube_node_status_condition{condition="Ready",job="kube-state-metrics",status="true"} * on (node) group_right() max(label_replace(ceph_disk_occupation{job="rook-ceph-mgr"},"node","$1","exported_instance","(.*)")) by (node)
+      record: cluster:ceph_node_down:join_kube
+    - expr: |
+        avg(max by(instance) (label_replace(label_replace(ceph_disk_occupation{job="rook-ceph-mgr"}, "instance", "$1", "exported_instance", "(.*)"), "device", "$1", "device", "/dev/(.*)") * on(instance, device) group_right() (irate(node_disk_read_time_seconds_total[1m]) + irate(node_disk_write_time_seconds_total[1m]) / (clamp_min(irate(node_disk_reads_completed_total[1m]), 1) + irate(node_disk_writes_completed_total[1m])))))
```

Comment:
Looks like this is rebased on an older commit. It's doing the change here, which should not be.

Reply:
reverted.